radicle-cli/src/terminal/io.rs

```diff
@@ -176,20 +176,22 @@ pub fn abort<D: fmt::Display>(prompt: D) -> bool {
 }
 
 /// Get the signer. First we try getting it from ssh-agent, otherwise we prompt the user.
-pub fn signer(profile: &Profile) -> anyhow::Result<Box<dyn Signer>> {
-    let signer = if let Ok(passphrase) = read_passphrase_from_env_var() {
-        MemorySigner::load(&profile.keystore, passphrase)?.boxed()
-    } else if let Ok(signer) = profile.signer() {
-        signer.boxed()
-    } else {
-        let passphrase = secret_input();
-        let spinner = spinner("Unsealing key...");
-        let signer = MemorySigner::load(&profile.keystore, passphrase)?;
+pub fn signer(profile: &Profile, interactive: bool) -> anyhow::Result<Box<dyn Signer>> {
+    if let Ok(passphrase) = read_passphrase_from_env_var() {
+        return Ok(MemorySigner::load(&profile.keystore, passphrase)?.boxed());
+    }
 
-        spinner.finish();
-        signer.boxed()
-    };
-    Ok(signer)
+    let signer = profile.signer();
+    if signer.is_ok() || !interactive {
+        return signer.map(|s| s.boxed()).map_err(|e| e.into());
+    }
+
+    let passphrase = secret_input();
+    let spinner = spinner("Unsealing key...");
+    let signer = MemorySigner::load(&profile.keystore, passphrase)?;
+
+    spinner.finish();
+    Ok(signer.boxed())
```

Comment:
This function is specifically designed for interactive use inside the cli, I don't think it makes sense to have a boolean change its behavior.. I'd just add a `passphrase_signer` function somewhere, and share that code with the remote helper.

Reply:
> This function is specifically designed for interactive use inside the cli, I don't think it makes sense to have a boolean change its behavior.. I'd just add a `passphrase_signer` function somewhere, and share that code with the remote helper.

Where would you recommend?  Would the new dependency on 'radicle-cli' be considered a problem?  Term::io appears best but not perfect because of its use of env vars for input/configuration.

> I'd just add a passphrase_signer function somewhere, and share that code with the remote helper.

I'll rename and split the function(s).